packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelFormFields/DefaultValueField.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useMemo } from 'react'
+import { ColumnTypeEnum } from '@sage-bionetworks/synapse-types'
+import {
+  FormControl,
+  MenuItem,
+  Select,
+  SelectProps,
+  TextField,
+  TextFieldProps,
+} from '@mui/material'
+import { getTextFieldType } from '../TableColumnSchemaEditorUtils'
+
+export type DefaultValueFieldProps<T> = {
+  columnType: ColumnTypeEnum
+  value: T
+  onChange: (newValue: T) => void
+  TextFieldProps?: Omit<TextFieldProps, 'value' | 'onChange'>
+  SelectProps?: Omit<SelectProps, 'value' | 'onChange'>
+}
+
+export default function DefaultValueField<T>(props: DefaultValueFieldProps<T>) {
+  const { columnType, onChange, value, TextFieldProps, SelectProps } = props
+
+  const textFieldType: TextFieldProps['type'] = useMemo(
+    () => getTextFieldType(columnType),
+    [columnType],
+  )
```

Comment:
Utility will update the `type` passed to the `input` element based on the column type (e.g. STRING -> 'text', DATE -> 'date', INTEGER -> 'number')